src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -508,6 +513,51 @@ private void validatePluginArguments() {
         // finally, give each plugin a chance to trim down any unseen instances from it's own list
         pluginDescriptors.entrySet().forEach(e -> e.getValue().validateArguments());
     }
+    /**
+     * Check the provided value against any range constraints specified in the Argument annotation
+     * for the corresponding field. Throw an exception if limits are violated.
+     *
+     * - Only checks numeric types (int, double, etc.)
+     */
+    private void checkArgumentRange(final ArgumentDefinition argumentDefinition, final Object argumentValue) {
+        // Only validate numeric types
+        if (!Number.class.isAssignableFrom(argumentDefinition.type)) {
+            return;
+        }
+
+        final Double argumentDoubleValue = (argumentValue == null) ? null : ((Number)argumentValue).doubleValue();
+
+        // Check hard limits first, if specified
+        if (isOutOfRange(argumentDefinition.minValue, argumentDefinition.maxValue, argumentDoubleValue)) {
+            throw new CommandLineException.OutOfRangeArgumentValue(argumentDefinition.fullName, argumentDefinition.minValue, argumentDefinition.maxValue, argumentValue);
+        }
+        // Check recommended values
+        if (isOutOfRange(argumentDefinition.minRecommendedValue, argumentDefinition.maxRecommendedValue, argumentDoubleValue)) {
+            final boolean outMinValue = argumentDefinition.minRecommendedValue != Double.NEGATIVE_INFINITY;
+            final boolean outMaxValue = argumentDefinition.maxRecommendedValue != Double.POSITIVE_INFINITY;
+            if (outMinValue && outMaxValue) {
+                logger.warn("Argument --{} has value {}, but recommended within range ({},{})",
```

Comment:
We should have test cases that exercise these code paths (I don't think there are any ?), especially if we're going to use the {} style formatting, to at least ensure that we don't blow up here.